spec/core/warning/warn_spec.rb

```diff
@@ -0,0 +1,137 @@
+require_relative '../../spec_helper'
+
+describe "Warning.warn" do
+  it "complains" do
+    -> {
+      Warning.warn("Chunky bacon!")
+    }.should complain("Chunky bacon!")
+  end
+
+  it "does not add a newline" do
+    ruby_exe("Warning.warn('test')", args: "2>&1").should == "test"
+  end
+
+  it "returns nil" do
+    ruby_exe("p Warning.warn('test')", args: "2>&1").should == "testnil\n"
+  end
+
+  it "extends itself" do
+    Warning.singleton_class.ancestors.should include(Warning)
+  end
+
+  it "has Warning as the method owner" do
+    ruby_exe("p Warning.method(:warn).owner").should == "Warning\n"
+  end
+
+  # NATFIXME: Implement IO#sync=
+  # it "can be overridden" do
+  #   code = <<-RUBY
+  #     $stdout.sync = true
+  #     $stderr.sync = true
+  #     def Warning.warn(msg)
+  #       if msg.start_with?("A")
+  #         puts msg.upcase
+  #       else
+  #         super
+  #       end
+  #     end
+  #     Warning.warn("A warning!")
+  #     Warning.warn("warning from stderr\n")
+  #   RUBY
+  #   ruby_exe(code, args: "2>&1").should == %Q[A WARNING!\nwarning from stderr\n]
+  # end
```

Comment:
`IO#sync=` is now implemented, this one can be reactivated.

Reply:
Done